source/Various/RootsOfBooleanFunctions.lagda

```diff
@@ -510,9 +510,163 @@ hypothesis to prove the correctness of ε-formula'. We did find such a
+  c₀ : E k
+  c₀ = (𝕗 ∘ f ∘ cons O) (εᵉ (𝕗 ∘ f ∘ cons O))
+
+  c₁ : E (succ n)
+  c₁ = (𝕗 ∘ cons O) (εᵉ (𝕗 ∘ cons O))
+
+  c₀-property : c₀ ＝ 𝕞𝕒𝕡 f c₁
+  c₀-property = (𝕗 ∘ f ∘ cons O) (εᵉ (𝕗 ∘ f ∘ cons O))        ＝⟨ I ⟩
+                (𝕗 ∘ f ∘ cons O) (𝕞𝕒𝕡s (f ∘ cons O) (εᵉ 𝕗))   ＝⟨ refl ⟩
+                𝕞𝕒𝕡 (f ∘ cons O) (𝕗 (εᵉ 𝕗))                   ＝⟨ II ⟩
+                𝕞𝕒𝕡 f (𝕞𝕒𝕡 (cons O) (𝕗 (εᵉ 𝕗)))               ＝⟨ refl ⟩
+                𝕞𝕒𝕡 f ((𝕗 ∘ cons O) ((𝕞𝕒𝕡s (cons O) (εᵉ 𝕗)))) ＝⟨ III ⟩
+                𝕞𝕒𝕡 f ((𝕗 ∘ cons O) (εᵉ (𝕗 ∘ cons O)))        ＝⟨ refl ⟩
+                𝕞𝕒𝕡 f c₁ ∎
+   where
+     I = ap (𝕗 ∘ f ∘ cons O) (unroll-εᵉ-lemma (f ∘ cons O))
+     II = 𝕞𝕒𝕡-𝕞𝕒𝕡 f O (𝕗 (εᵉ 𝕗)) ⁻¹
+     III = ap (𝕞𝕒𝕡 f ∘ (𝕗 ∘ cons O)) (unroll-εᵉ-lemma (cons O) ⁻¹)
+
+  γ :  εᵉ (𝕗 ∘ f) ＝ 𝕞𝕒𝕡s f (εᵉ 𝕗)
+  γ = εᵉ (𝕗 ∘ f) ＝⟨ refl ⟩
+      c₀ , (εᵉ (𝕗 ∘ f ∘ cons c₀))                  ＝⟨ I ⟩
+      𝕞𝕒𝕡 f c₁ , (εᵉ (𝕗 ∘ f ∘ cons (𝕞𝕒𝕡 f c₁)))    ＝⟨ II ⟩
+      𝕞𝕒𝕡 f c₁ , 𝕞𝕒𝕡s (f ∘ cons (𝕞𝕒𝕡 f c₁)) (εᵉ 𝕗) ＝⟨ III ⟩
+      𝕞𝕒𝕡 f c₁ , 𝕞𝕒𝕡s f (𝕞𝕒𝕡s (cons c₁) (εᵉ 𝕗))    ＝⟨ IV ⟩
+      𝕞𝕒𝕡 f c₁ , 𝕞𝕒𝕡s f (εᵉ (𝕗 ∘ cons c₁))         ＝⟨ refl ⟩
+      𝕞𝕒𝕡s f (c₁ , (εᵉ (𝕗 ∘ cons c₁)))             ＝⟨ refl ⟩
+      𝕞𝕒𝕡s f (εᵉ 𝕗) ∎
+   where
+    I = ap (λ x → x , (εᵉ (𝕗 ∘ f ∘ cons x))) c₀-property
+    II = ap (𝕞𝕒𝕡 f c₁ ,_) (unroll-εᵉ-lemma (f ∘ cons (𝕞𝕒𝕡 f c₁)))
+    III = ap (𝕞𝕒𝕡 f c₁ ,_) (𝕞𝕒𝕡-𝕞𝕒𝕡s f c₁ (εᵉ 𝕗) ⁻¹)
+    IV = ap (λ x → 𝕞𝕒𝕡 f c₁ , 𝕞𝕒𝕡s f x) (unroll-εᵉ-lemma (cons c₁) ⁻¹)
+
+unroll-εᵉ : {n : ℕ}
+          → (e₀ : E (succ n))
+          → εᵉ (𝕗 ∘ (cons e₀)) ＝ 𝕔𝕠𝕟𝕤s e₀ (εᵉ 𝕗)
+unroll-εᵉ e₀ = unroll-εᵉ-lemma (cons e₀) ∙ 𝕞𝕒𝕡s-cons-𝕔𝕠𝕟𝕤s e₀ (εᵉ 𝕗)
+
+\end{code}
+
+From there we can show that ε-formula' n and ε-formula n are indeed
+equals.
+
+\begin{code}
+
+formulas-are-equal : (n : ℕ) → ε-formula' n ＝ ε-formula n
+formulas-are-equal 0 = refl
+formulas-are-equal (succ n) = γ
+ where
+  c₀ : E (succ n)
+  c₀ = (𝕗 ∘ cons O) (εᵉ (𝕗 ∘ cons O))
+
+  c₁ : E (succ n)
+  c₁ = (𝕗 ∘ cons O) (𝕔𝕠𝕟𝕤s O (ε-formula n))
+
+  c₀-property : c₀ ＝ c₁
+  c₀-property = (𝕗 ∘ cons O) (εᵉ (𝕗 ∘ cons O))  ＝⟨ I ⟩
+                (𝕗 ∘ cons O) (𝕔𝕠𝕟𝕤s O (εᵉ 𝕗))   ＝⟨ II ⟩
+                (𝕗 ∘ cons O) (𝕔𝕠𝕟𝕤s O (ε-formula n)) ∎
+   where
+    I = ap (𝕗 ∘ cons O) (unroll-εᵉ O)
+    II = ap (𝕗 ∘ cons O ∘ 𝕔𝕠𝕟𝕤s O) (formulas-are-equal n)
+
+  γ : ε-formula' (succ n) ＝ ε-formula (succ n)
+  γ = ε-formula' (succ n)            ＝⟨ refl ⟩
+      εᵉ 𝕗                           ＝⟨ refl ⟩
+      c₀ , εᵉ (𝕗 ∘ cons c₀)          ＝⟨ I ⟩
+      c₀ , (𝕔𝕠𝕟𝕤s c₀ (ε-formula' n)) ＝⟨ refl ⟩
+      c₀ , (𝕔𝕠𝕟𝕤s c₀ (εᵉ 𝕗))         ＝⟨ II ⟩
+      c₀ , (𝕔𝕠𝕟𝕤s c₀ (ε-formula n))  ＝⟨ III ⟩
+      c₁ , (𝕔𝕠𝕟𝕤s c₁ (ε-formula n))  ＝⟨ refl ⟩
+      ε-formula (succ n) ∎
+   where
+    I = ap (c₀ ,_) (unroll-εᵉ c₀)
+    II = ap (λ x → c₀ , (𝕔𝕠𝕟𝕤s c₀ x)) (formulas-are-equal n)
+    III = ap (λ x → x , (𝕔𝕠𝕟𝕤s x (ε-formula n))) c₀-property
+\end{code}
+
+It then follow immediatly the correctness of ε-formula' n by
```

Comment:
"follow" -> "follows".

"immediatly" -> "immediately".

You probably want to rephrase this as "It then follows immediately by transport that ε-formula' is correct".